yarn/pom.xml

```diff
@@ -92,12 +92,28 @@
       <artifactId>jetty-servlet</artifactId>
     </dependency>
     <!-- End of shaded deps. -->
-    
+
+    <!--
+      SPARK-10059: Explicitly add JSP dependencies for tests since the MiniYARN cluster needs them.
+    -->
+    <dependency>
+      <groupId>org.eclipse.jetty.orbit</groupId>
+      <artifactId>javax.servlet.jsp</artifactId>
+      <version>2.2.0.v201112011158</version>
```

Comment:
Oh now I remember. JSP is also versioned separately such that 2.2 is part of Servlet 3.0. Carry on. http://tomcat.apache.org/whichversion.html